guidance/guidance_los/guidance_los/los_guidance_algorithm.py

```diff
@@ -0,0 +1,114 @@
+#!/usr/bin/env python3
+
+import numpy as np
+
+
+class ThirdOrderLOSGuidance:
+    """Line-of-Sight (LOS) guidance algorithm."""
+
+    def __init__(self, config_dict: dict):
+        # Initialize parameters
+        self.h_delta_min = config_dict['h_delta_min']
+        self.h_delta_max = config_dict['h_delta_max']
+        self.h_delta_factor = config_dict['h_delta_factor']
+        self.nominal_speed = config_dict['nominal_speed']
+        self.min_speed = config_dict['min_speed']
+        self.max_pitch_angle = config_dict['max_pitch_angle']
+        self.depth_gain = config_dict['depth_gain']
+
+        # Filter parameters
+        self.x = np.zeros(9)
+        self.omega = np.diag(config_dict['filter']['omega_diag'])
+        self.zeta = np.diag(config_dict['filter']['zeta_diag'])
+
+        # Initialize filter matrices
+        self.setup_filter_matrices()
+
+    def setup_filter_matrices(self):
+        self.Ad = np.zeros((9, 9))
+        self.Bd = np.zeros((9, 3))
+
+        # Fill Ad matrix blocks
+        self.Ad[0:3, 3:6] = np.eye(3)
+        self.Ad[3:6, 6:9] = np.eye(3)
+
+        # Compute characteristic polynomial coefficients
+        omega_cubed = self.omega @ self.omega @ self.omega
+        omega_squared = self.omega @ self.omega
+
+        # Fill in the acceleration dynamics
+        self.Ad[6:9, 0:3] = -omega_cubed
+        self.Ad[6:9, 3:6] = -(2 * self.zeta + np.eye(3)) @ omega_squared
+        self.Ad[6:9, 6:9] = -(2 * self.zeta + np.eye(3)) @ self.omega
+
+        # Input matrix
+        self.Bd[6:9, :] = omega_cubed
+
+    def ssa(self, angle: float) -> float:
+        return (angle + np.pi) % (2 * np.pi) - np.pi
```

Comment:
the ssa method could be a static method instead, since it does not rely on self.
```python
@staticmethod
def ssa(self, angle: float) -> float:
    return (angle + np.pi) % (2 * np.pi) - np.pi